service/src/main/java/uk/gov/hmcts/reform/fpl/model/DirectionResponse.java

```diff
@@ -21,4 +22,51 @@
     private final String cannotComplyReason;
     private final List<String> c2Uploaded;
     private final DocumentReference cannotComplyFile;
+    @JsonIgnore
+    private String respondingOnBehalfOf;
+
+    @JsonGetter("respondingOnBehalfOfRespondent")
+    private String getRespondingOnBehalfOfRespondent() {
+        if (respondingOnBehalfOf != null && respondingOnBehalfOf.contains("RESPONDENT")) {
+            return respondingOnBehalfOf;
+        }
+        return null;
+    }
+
+    @JsonGetter("respondingOnBehalfOfOthers")
+    private String getRespondingOnBehalfOfOthers() {
+        if (respondingOnBehalfOf != null && respondingOnBehalfOf.contains("OTHER")) {
+            return respondingOnBehalfOf;
+        }
+        return null;
+    }
+
+    @JsonGetter("respondingOnBehalfOfCafcass")
+    private String getRespondingOnBehalfOfCafcass() {
+        if (respondingOnBehalfOf != null && respondingOnBehalfOf.contains("CAFCASS")) {
+            return respondingOnBehalfOf;
+        }
+        return null;
+    }
+
+    @JsonSetter("respondingOnBehalfOfRespondent")
```

Comment:
When Jackson sees `respondingOnBehalfOfRespondent` property in JSON it will try to find either field or setter that it can use in deserliasation process.